substratevm/src/com.oracle.svm.jni/src/com/oracle/svm/jni/functions/JNIFunctions.java

```diff
@@ -378,7 +378,11 @@ static int RegisterNatives(JNIEnvironment env, JNIObjectHandle hclazz, JNINative
                                     MetaUtil.internalNameToJava(l.getDeclaringClassName(), true, false) +
                                     "." + l.getName() + l.getDescriptor() + ")";
                 }
-                throw new NoSuchMethodError(message);
+                /*
+                 * If during analysis we didn't link this JNI method, this doesn't mean that we
+                 * should abort loading of the native library by throwing NoSuchMethodException.
```

Comment:
Really? I would think that a missing JNI registration at image build time is a problem. Also, what should a user do with the printed warning message? It most likely means that the JNI library is not working properly anyway.

@peter-hofer ?

Reply:
It happens that libraries register arbitrary Java native methods from their native code. If during analysis, we didn't reach some of those JNI methods (see https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.jni/src/com/oracle/svm/jni/hosted/JNINativeCallWrapperSubstitutionProcessor.java#L42-L45  and https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.jni/src/com/oracle/svm/jni/access/JNIAccessFeature.java#L225-L250) we shouldn't fail: those native methods can never be invoked. 

About this printed warning message, I didn't see another way how to do that, any suggestions?

Reply:
Ah I see, that is to register native methods when calling from Java to C, and not for calling from C to Java. So in that case we can then just silently ignore the methods. Why print a warning message when we know that the method is not going to be invoked?

Reply:
All right, I'll delete this printing warning message.